content/post/2025/02-riscv-formal-extension-for-vc-formal.md

```diff
@@ -0,0 +1,298 @@
+
+The framework consists of **three key stages**, each automated for smooth verification:  
+
+1. **Pre-Processing:** Converts `.sby` files into `.tcl` scripts for **VC Formal compatibility**.  
+2. **Processing:** Runs the **formal checks**, collects logs, and categorizes **warnings & errors**.  
+3. **Post-Processing:** Summarizes verification **results**, helping users quickly identify pass/fail conditions.  
+
+To achieve this, we’ve extended the framework with four core components:  
+
+### **1️⃣ sby_to_tcl.py – Automating Test Setup (Pre-Processing)**  
+The `sby_to_tcl.py` script automates the **conversion of SymbiYosys `.sby` files** into **VC Formal-compatible `.tcl` scripts**. It also **creates a Makefile** that organizes multiple verification tasks, allowing batch execution with a single command.  
+
+#### 🔹 **Why is this important?**  
+- Eliminates **manual effort** in handling `.tcl` files.  
+- Enables **batch verification** by processing multiple `.sby` files at once.  
+- Organizes output systematically in a dedicated `vcf` directory.  
+
+### **2️⃣ vcf_cexdata.sh – Error & Warning Processing (Processing Stage)**  
+During the verification execution, the `vcf_cexdata.sh` script ensures **structured handling of warnings and errors**. Instead of overwriting logs, it **captures and organizes logs** for all checks, ensuring **every warning/error is properly recorded**.  
+
+#### 🔹 **Key Features:**  
+- Prevents loss of information by **storing logs separately**.  
+- Formats errors in an **easy-to-read structure** (`warnings.txt`, `errors.txt`).  
+- Provides a **quick summary** of issues across multiple checks.  
+
+### **3️⃣ vcf_res_process.py – Summarizing Verification Results (Post-Processing)**  
+After verification, `vcf_res_process.py` extracts key results from each check, categorizing assertions as **PASS, FAIL, or INCONCLUSIVE** and consolidating everything into a **single summary file (`results.txt`)**.  
+
+#### 🔹 **Why does this matter?**  
+- **Saves time**—no need to inspect multiple files manually.  
+- Provides a **high-level overview** of all formal verification checks.  
+- Helps users **quickly identify issues** that need further debugging.  
+
+### **4️⃣ Makefile – Automating the Entire Workflow**  
+The **Makefile** serves as the backbone of automation, orchestrating all three stages. With simple commands like `make vcf_check`, users can **run all checks**, collect results, and review summaries **without manually executing scripts**.  
+
+#### 🔹 **Key Features:**  
+- **One-command execution** of all verification steps.  
+- **Automated log collection & clean-up** for an organized workflow.  
+- **Results displayed directly in the terminal** for quick review.  
+
+For a **more detailed breakdown** on each file/process, check out the complete repository on **[GitHub](https://github.com/Chaotic-VRBlue/riscv-formal-vc-formal-extension?tab=readme-ov-file#file-descriptions)**! 🚀  
```

Comment:
I would change the link text to "complete repository on GitHub" instead of just "GitHub" for accessibility reasons (you're linking to the repository, not to github itself).  I also don't think it needs to be bolded, the link formatting is enough emphasis.